nvmeadm/src/nvmf_subsystem.rs

```diff
@@ -243,15 +243,23 @@ impl Subsystem {
 
     /// Returns the particular subsystem based on the nqn and address.
     // TODO: Optimize this code.
-    pub fn get(host: &str, port: &u16, nqn: &str) -> Result<Subsystem, NvmeError> {
+    pub fn get(
+        host: &str,
+        port: &u16,
+        transport: TrType,
+        nqn: &str,
+    ) -> Result<Subsystem, NvmeError> {
         let nvme_subsystems = NvmeSubsystems::new()?;
 
         let host = host.to_string();
         let sport = port.to_string();
-        match nvme_subsystems
-            .flatten()
-            .find(|subsys| subsys.nqn == *nqn && subsys.address.match_host_port(&host, &sport))
-        {
+        match nvme_subsystems.flatten().find(|subsys| {
+            subsys.nqn == *nqn
+                && subsys.address.match_host_port(&host, &sport)
+                && subsys
+                    .transport
+                    .eq_ignore_ascii_case(transport.to_string().as_str())
```

Comment:
You can conversion above to avoid converting to string on every iteration

Reply:
done.